pkg/generator/rps.go

```diff
@@ -0,0 +1,142 @@
+package generator
+
+import (
+	"fmt"
+	"github.com/vhive-serverless/loader/pkg/common"
+	"github.com/vhive-serverless/loader/pkg/config"
+	"math"
+	"math/rand"
+)
+
+func generateFunctionByRPS(experimentDuration int, rpsTarget float64) (common.IATArray, []int) {
+	iat := 1000000.0 / float64(rpsTarget) // μs
+
+	var iatResult []float64
+	var countResult []int
+
+	duration := 0.0 // ms
+	totalExperimentDurationMs := float64(experimentDuration * 60_000_000.0)
+
+	currentMinute := 0
+	currentCount := 0
+
+	for duration < totalExperimentDurationMs {
+		iatResult = append(iatResult, iat)
+		duration += iat
+		currentCount++
+
+		if int(duration)/60_000_000 != currentMinute {
+			countResult = append(countResult, currentCount)
+
+			currentMinute++
+			currentCount = 0
+		}
```

Comment:
Didn't get that. Is it executed almost always? currentMinute is incremented by one, duration - by iat, they never will result in false condition. Except the case of iat being the full minute, then it becomes completely weird.

Reply:
No. The purpose of this is to count the number of invocations within a minute, i.e. to generate the `countResult` array.

Reply:
Ah, I see, that's integer division. I still propose to be clearer with the inequality signs and replace it with <. The result should be the same.